presto-parser/src/test/java/com/facebook/presto/sql/parser/TestSqlParser.java

```diff
@@ -1496,6 +1498,29 @@ public void testCreateFunction()
                 "Duplicate null-call clause: CALLEDONNULLINPUT");
     }
 
+    @Test
+    public void testAlterFunction()
+    {
+        QualifiedName functionName = QualifiedName.of("testing", "default", "tan");
+        assertStatement(
+                "ALTER FUNCTION testing.default.tan\n" +
+                        "CALLED ON NULL INPUT",
+                new AlterFunction(functionName, Optional.empty(), new AlterRoutineCharacteristics(Optional.of(CALLED_ON_NULL_INPUT))));
+        assertStatement(
+                "ALTER FUNCTION testing.default.tan(double)\n" +
+                        "RETURNS NULL ON NULL INPUT",
+                new AlterFunction(functionName, Optional.of(ImmutableList.of("double")), new AlterRoutineCharacteristics(Optional.of(RETURNS_NULL_ON_NULL_INPUT))));
+
+        assertInvalidStatement(
+                "ALTER FUNCTION testing.default.tan",
+                "No alter routine characteristics specified");
+        assertInvalidStatement(
+                "ALTER FUNCTION testing.default.tan\n" +
+                        "RETURNS NULL ON NULL INPUT\n" +
+                        "RETURNS NULL ON NULL INPUT",
+                "Duplicate null-call clause: RETURNSNULLONNULLINPUT");
```

Comment:
Does it have to print like this 🤣

Reply:
For now, it is the print out. Maybe I can find a way to alter it in the future. :)